nix/modules/environment.nix

```diff
@@ -0,0 +1,59 @@
+{ lib, config, pkgs, ... }:
+
+{
+  options.environment = {
+    systemPackages = lib.mkOption {
+      type = lib.types.listOf lib.types.package;
+      default = [ ];
+    };
+  };
+
+  config =
+    let
+      pathDir = "/run/system-manager/sw";
```

Comment:
Maybe this should be `/run/current-system/sw` for NixOS compatibility?

Reply:
Yeah, though that would preclude ever running system-manager on nixos. I don't suppose that that would be a very useful thing in any case, but I'm not entirely sure that no-one will find some odd use-case.
We could make yet another option to configure this and default to `current-system` though.